src/platform/plugins/shared/dashboard/server/content_management/v3/transforms/out/control_state.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { defaults, flow } from 'lodash';
+import { SerializableRecord } from '@kbn/utility-types';
+import { DEFAULT_CONTROL_GROW, DEFAULT_CONTROL_WIDTH } from '@kbn/controls-plugin/common';
+import { ControlGroupAttributes } from '../../types';
+
+/**
+ * Transform functions for serialized controls state.
+ */
+export const transformControlsState: (
+  serializedControlState: string
+) => ControlGroupAttributes['controls'] = flow(
+  JSON.parse,
```

Comment:
I wonder if there is any way to give `flow` explicit types - because right now it seems to pass implicit `any`s which need to be typed in each individual function.

It isn't necessary to fix this now, but IMO this leaves room for errors because the output of one function isn't guaranteed to be the type that the next function takes. I wonder how RXJS' `pipe` function handles its strict typing.

Reply:
I think we'd have to add a bunch of one-off types to provide to flow. 

I've seen core using the similar [fp-ts library](https://gcanti.github.io/fp-ts/) which seems to infer types. But that whole library seems like overkill. I'd rather just write the functions like `transformC(transformB(transformA(input)))`. 😆